### ISSUE TITLE
Streaming message stops it's streaming request in deployed version of the app

### ISSUE THREAD

**Issue by akshat-OwO** [joschan21/quill#8]
The app works fine in local production build, but not in vercel. When you ask a question getFileMessages request interrupts the message stream response resulting in no output

I have deployed my app here https://down-one.vercel.app

**Comment by emptypockets-dev** [comment]
I think I am having this issue as well but cannot pin it down to `getFileMessages`. For some responses, it will start to provide a sentence or two and then disappear without finishing.

**Comment by emptypockets-dev** [comment]
@akshat-OwO how do you know that "getFileMessages request interrupts the message stream response resulting in no output"? Is there any way I can confirm?

**Comment by akshat-OwO** [comment]
You can check the network tab in the developer console. The `getFileMessages` request runs without letting the `message ` request finish so that the AI replied message can be added to the database. 
Henceforth, the ai message is not being added in the database which results in `getFileMessages` fetching the previous messages without the ai message

**Comment by emptypockets-dev** [comment]
Yea, that does appear to be what's happening. Can confirm.

**Comment by emptypockets-dev** [comment]
Any idea why this only happens on the deployed version? (hoping that provides some insight into how to fix)

I checked the local and deployed versions have the same requests but don't "interrupt"

**Comment by akshat-OwO** [comment]
yeah the problem is only in deployed version of the app

**Comment by emptypockets-dev** [comment]
Right, but why? what's the difference between how it runs locally and in the deployed version? I've tried to dig in but haven't really been able to figure it out. My best guess is that something about the way `onSettled` runs Is the cause. 
```
    onSettled: async () => {
      setIsLoading(false);
      await utils.getFileMessages.invalidate({ fileId });
    },
``` 

I don't quite understand why we're invalidating here. Any idea? If I comment that out then the messages that come back through are out of order.

**Comment by philip06lee** [comment]
The app works fine in local but after deploying into vercel, the response to question is like not connected to pinecone.

**Comment by anand-mukul** [comment]
![pdfninja](https://github.com/joschan21/quill/assets/147015644/7b7b770f-102d-4185-b30e-abf6ee74c136)

When I ask it to write an essay, it only writes a few lines and then stops working. But working fine in short response.

**Comment by philip06lee** [comment]
When the answer is too long, it appears a short while but then never get posted. Is it because there is a  limit on how long the answer should be? So need to prompt please answer in 3 short sentences for example, then the answer is posted.

**Comment by rishav887** [comment]
did anyone got the fix? i have not been able to reach the cause

**Comment by rishav887** [comment]
@anand-mukul Dint understand what you pointing me to. i am not getting the pdf issue.The messages are getting interrupted in between and then disappearing.

**Comment by anand-mukul** [comment]
@rishav887 There is a misunderstanding; that's not for you, and it doesn't mean I'm pointing you to that issue.

**Comment by philip06lee** [comment]
Could it be in the prompts ? I tried with different prompts for deployed version. If you add "answer concisely", the answer will get posted.

**Comment by rishav887** [comment]
in that case shouldnt it fail on the local as well ?
@joschan21 any advise here? bringing this to your notice.

**Comment by rishav887** [comment]
@joschan21 any advise here? bringing this to your notice.

**Comment by salimi-my** [comment]
I maybe wrong, but I think this issue cause by Vercel. I checked logs of /api/message on Vercel and it shows Execution Duration / Limit 10.01s / 10s (timed out).

![image](https://github.com/joschan21/quill/assets/55532224/0c0e7874-57cb-4fec-adf2-50fd6ea96bb9)

This limit of 10s execution duration explain why short responses works but does not work for long responses and why these issue only happened on production but not on local.

![image](https://github.com/joschan21/quill/assets/55532224/4d408502-eef0-4f1e-b332-8f05beed29c2)

The only solution is to upgrade your Hobby plan to Pro plan or deploy on other platforms.

**Comment by rishav887** [comment]
@salimi-my Thanks! i guess thats the issue. any suggested workarounds ?

**Comment by stabildev** [comment]
You can use a Cloudflare worker as your OpenAI gateway but it's a LOT of work with many pitfalls. Your ChatContext.tsx mutation function will then make the request to you CF worker (be sure to include auth cookies from Kinde and take care of CORS). The worker then requests the prompt from your backend, using the forwarded Kinde auth cookies. The worker then makes the request to OpenAI and forwards the response back to the waiting client (in ChatContext). When the stream finishes the worker makes an additional request with the completed message to another backend endpoint to save the message in the db.

**Comment by philip06lee** [comment]
Noticed that for large size pdf, say 80 pages, the streaming message can also stopped while nearing the end, even in local version. Think this has to do with GPT3.5 when task gets bigger, bigger token is needed. Perhaps if linked to GPT4, this would not happen. Anyone has any comment?

**Comment by baghelkunalpal** [comment]
@akshat-OwO  bro im not able to chat with pdf Too many pages are there that page show only actually i m not able to type something in input box

**Comment by akshat-OwO** [comment]
@baghelkunalpal There can be many reasons why you are getting this error. You can take a look at my [repo](https://github.com/akshat-OwO/down) although the app is not in the functioning state as I ran out of openai credits but I was able to complete this project.

**Comment by baghelkunalpal** [comment]
> You can check the network tab in the developer console. The `getFileMessages` request runs without letting the `message ` request finish so that the AI replied message can be added to the database. Henceforth, the ai message is not being added in the database which results in `getFileMessages` fetching the previous messages without the ai message
i think your application work fine i need you help bcz its my major project in college there is two issue there like im not able to chat and subcription based 
![image](https://github.com/joschan21/quill/assets/126814924/f307be92-ff25-4945-bd77-a2d0d00b4257)

**Comment by baghelkunalpal** [comment]
> @baghelkunalpal There can be many reasons why you are getting this error. You can take a look at my [repo](https://github.com/akshat-OwO/down) although the app is not in the functioning state as I ran out of openai credits but I was able to complete this project.

i m using new email for the openai i have $5
